tests/cli-v2.rs

```diff
@@ -1185,3 +1185,36 @@ fn test_complete_profile_skips_missing_when_forced() {
         );
     })
 }
+
+#[test]
+fn run_with_install_flag_against_unavailable_component() {
+    setup(&|config| {
+        let trip = TargetTriple::from_build();
+        make_component_unavailable(config, "rust-std", &trip);
+        expect_ok_ex(
+            config,
+            &[
+                "rustup",
+                "run",
+                "--install",
+                "nightly",
+                "rustc",
+                "--version",
+            ],
+            "1.3.0 (hash-nightly-2)
+",
+            for_host!(
+                r"info: syncing channel updates for 'nightly-{0}'
+info: latest update on 2015-01-02, rust version 1.3.0 (hash-nightly-2)
+warning: Force-skipping unavailable component 'rust-std-x86_64-apple-darwin'
```

Comment:
Unfortunately I imagine this will also cause problems.  You need:
```suggestion
warning: Force-skipping unavailable component 'rust-std-{0}'
```
most likely